### ISSUE TITLE
Lists should autocomplete on return

### ISSUE THREAD

**Issue by markmals** [lukakerr/twig#16]
When you start a [list](https://daringfireball.net/projects/markdown/syntax#list) (using `*`, `-`, `+`, or a number followed by a period), type an item, and hit return the editor should (at least have the option to) autocomplete a new list element for the next line.

For instance, if I type `* Milk` and hit return, the next line should immediately have a `*` and a `space` so I can easily continue adding to the list. Same for ordinal lists; the next line should have the next number in sequential order.

Additionally, if I hit return again, it should remove the `* space` it inserted and return again, discontinuing the list.

**Comment by alexmsenger** [comment]
Not sure if I should open a new issue or not, but as it seems related I'll leave it here:

I'd love to be able to change the list level (indent / outdent) with a <kbd>Tab</kbd>. The way I'm imagining it is: I select two or more lines, hit <kbd>Tab</kbd> and the selected lines will be prepended a `tab` character (or n spaces, depending on the settings possibly). When I hit <kbd>Shift</kbd>+<kbd>Tab</kbd> and the line begins with a `tab` character (or spaces), they should be removed. 

Let me know if I should open a separate issue for that. Thanks for the amazing product!

**Comment by lukakerr** [comment]
I agree with both comments, the autocomplete/keyboard shortcuts around lists needs to be improved. I'll put these features on my todo list.

**Comment by trueschool** [comment]
Thank you, yes. I'd like to see the autocomplete work on return, as well as easy keystrokes to turn selected text into an ordered or unordered list. On my old markdown editor MacDown, it was ⌘+shift+O or ⌘+shift+U. Thanks for considering. BTW this editor shows great promise in its current state! Looking forward to seeing updates.
